core/src/main/scala/org/apache/spark/Heartbeater.scala

```diff
@@ -45,7 +45,8 @@ private[spark] class Heartbeater(
     val heartbeatTask = new Runnable() {
       override def run(): Unit = Utils.logUncaughtExceptions(reportHeartbeat())
     }
-    heartbeater.scheduleAtFixedRate(heartbeatTask, initialDelay, intervalMs, TimeUnit.MILLISECONDS)
+    heartbeater.scheduleWithFixedDelay(
+      heartbeatTask, initialDelay, intervalMs, TimeUnit.MILLISECONDS)
```

Comment:
Can we revert this ? Other state piggybacks on heartbeat payload and even if there is jitter, it is better to have a regular/predictable cadence of heartbeat messages.

Reply:
Done